### PR TITLE
Handshake integration

### PR DESCRIPTION
**Pull request by coot**
This PR is to `coot/mux-application` branch (#515), only to reduce it's scope.  I hope :D #515 will be merged first.

### CODE REVIEW

File: typed-protocols/src/Network/TypedProtocol/Driver.hs [hunk 1]
```diff
@@ -286,6 +323,38 @@ runDecoderWithChannel Channel{recv} = go
...
+  deriving (Show)
+
+instance Exception failure => Exception (NonIncrementalDecoderFailure failure)
+
+
+-- |
+-- Like @'runDecoderWithChannel'@, but it allows to receive data from channel at
+-- most once.
+--
+runDecoderWithChannelNonIncremental
```
[the remark this answers is not in the document]
Comment (coot) [reply typed-protocols/src/Network/TypedProtocol/Driver.hs]: Hmm, you're right, this needs to be done on a different level.

File: ouroboros-network/src/Ouroboros/Network/Socket.hs [hunk 2]
```diff
@@ -51,6 +63,15 @@ import           Ouroboros.Network.Mux.Interface ( MuxApplication
 
 import           Text.Printf
 
+-- |
+-- We assume that a TCP segment size of 1440 bytes with initial window of size
+-- 4.  This sets upper limit of 5760 bytes on each message of handshake
+-- protocol.  If the limit is exceeded, then @'NotEnoughInput'@ exception will
+-- be thrown.
```
[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Socket.hs]: Maybe, depending on the context; from the point of view of `Codec` it's not inough input.

File: typed-protocols/src/Network/TypedProtocol/Driver.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (coot) [reply typed-protocols/src/Network/TypedProtocol/Driver.hs]: @avieth I changed this bit.  There is now `ByteChannel` in `Network.TypedProtocols.ByteChannel`, and a function that allows to run a `Peer` against it with an assumption that each message has a given upper bound of bytes.

[the remark this answers is not in the document]
Comment (avieth) [reply typed-protocols/src/Network/TypedProtocol/Driver.hs]: Shouldn't that be a part of the codec definition instead?

File: ouroboros-network/src/Ouroboros/Network/Channel.hs [hunk 3]
```diff
@@ -102,8 +103,8 @@ withFifosAsChannel fifoPathRead fifoPathWrite action =
 -- | Make a 'Channel' from a 'Socket'. The socket must be a stream socket
 --- type and status connected.
 ---
-socketAsChannel :: Socket.Socket -> Channel IO LBS.ByteString
-socketAsChannel socket =
+socketAsChannel' :: Int -> Socket.Socket -> Channel IO LBS.ByteString
```
[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Channel.hs]: Removed that commit.

File: typed-protocols/src/Network/TypedProtocol/Driver.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (coot) [reply typed-protocols/src/Network/TypedProtocol/Driver.hs]: It's `runPeer` like [function](https://github.com/input-output-hk/ouroboros-network/pull/537/files#diff-20a491028de67de5575b71234ae070c0R68).

**Comment by coot** [comment]
Closing this PR, since it needs to updated to `coot/mux-application-2` branch.

**Pull request closed without merging**